vwifi.c

```diff
@@ -91,12 +106,16 @@ struct owl_vif {
         /* Structure for AP mode */
         struct {
             bool ap_enabled;
+            bool privacy;
             /* List node for storing AP (owl->ap_list is the head),
              * this field is for interface in AP mode.
              */
             struct list_head ap_list;
         };
     };
+
+    u32 beacon_ie_len;
+    u8 beacon_ie[IE_MAX_LEN];
```

Comment:
Leave these two fields outside the union.

I found that there may be some bugs when we use union (inside `struct owl_vif`) to split the fields between STA and HostAP. For example, the line 668 if (`vif->scan_request`) { in `owl_scan()` is dangerous because if the `vif` is operating in HostAP mode, this line will read the memory area of the AP field. Off course we can avoid this by checking the operating mode in the beginning of every function, but sometimes we still forget to do so.

In conclusion, I think the union make our code prone to bugs. We should no longer use it.

Reply:
> In conclusion, I think the union make our code prone to bugs. We should no longer use it.

Send another pull request?

Reply:
Yes, I will send another PR latter.